baseapp/abci.go

```diff
@@ -122,7 +125,15 @@ func (app *BaseApp) SetOption(req abci.RequestSetOption) (res abci.ResponseSetOp
 // Info implements the ABCI interface.
 func (app *BaseApp) Info(req abci.RequestInfo) abci.ResponseInfo {
 	lastCommitID := app.cms.LastCommitID()
-
+	// load the app version for a non zero height
+	if lastCommitID.Version > 0 {
+		ctx, err := app.createQueryContext(lastCommitID.Version, false)
+		if err != nil {
+			panic(err)
+		}
+		// get and set the app version
+		_ = app.AppVersion(ctx)
```

Comment:
[question] where or how does this "set" the app version? The invoked function returns the app version but I don't know what "set" is referring to here.

https://github.com/celestiaorg/cosmos-sdk/blob/4c897c6e15c7c0b956ca9fa25d0625ff7a39553a/baseapp/baseapp.go#L208-L211